### ISSUE TITLE
Address contains no value -- in module (inet::Udp) 

### ISSUE THREAD

**Issue by AlessandroSpallina** [Unipisa/Simu5G#46]
Hi all, I'm unable to understand what's wrong with my code, I'm trying to use Veins and a MEC at the same time: when the first car sends a packet to the MEC I have this error message: 
```
Address contains no value -- in module (inet::Udp) DummyScenario.car[0].udp (id=519), at t=1.262s, event #11335
```
In order to reproduce this bug, you can find my simulation here: https://github.com/AlessandroSpallina/SyntheticTraceGenerator/tree/master/simulations/Villarosa

Any help with that?

I'm also trying to use the Monaco mobility model but I don't see any vehicle neither in SUMO nor in Omnetpp, the simulation for that is here: https://github.com/AlessandroSpallina/SyntheticTraceGenerator/tree/master/simulations/Monaco

Let me know if you have a tip jar for the help

**Comment by pjnaraujo-uminho** [comment]
Hello, I'm having the same exact problem trying to use Veins and the MEC resources of Simu5g. The same error is thrown when a Car wants to send the first packet to the ME Host. I also don't understand what the issue is - may be something related to the use of Car instead of Ue as it is in the provided simulation examples (?).

**Comment by giovanninardini** [comment]
Hello, I am sorry for the late response.

I guess you both are using Simu5G 1.2 with INET 4.3.2. 
In that case, INET 4.3.2 has a bug within the `HostAutoConfigurator` module, that is the module that assigns the IP address to dynamically-created modules (the cars, in your example).
When using MEC, the car has (at least) two applications, one of which is the `DeviceApp`. The latter communicates with other local apps using the loopback interface, which does not correctly get an IP address because of the bug I was mentioning above.

So, we informed INET developers about this and they promptly fixed that problem in INET 4.4. 
Now, you can do two things:
**1)** download the latest GitHub version of Simu5G, which now works with OMNeT 6.0 and INET 4.4, and use that version as a base for your scenario. 
**2)** manually fix the INET code.

**If you go for 1)**, it is likely that your own code must be updated to work with OMNeT 6.0 and INET 4.4, too. This might requires some non-trivial effort, but eventually your code will "live longer".

**If you go for 2)**, it is (hopefully) a quick fix... but at some point you may need to update anyway your code to newer versions of Simu5G/OMNeT/INET. 
For this solution, go to _inet/networklayer/configurator/ipv4/HostAutoConfigurator.cc_, line 87, and change the content of the `if` block as follows:

```
ipv4Data->setIPAddress(Ipv4Address::LOOPBACK_ADDRESS);
ipv4Data->setNetmask(Ipv4Address::LOOPBACK_NETMASK);
ipv4Data->setMetric(1);
```
Also, move the initialization of `auto ipv4Data` before the `if` block.

Hope this helps.

Best regards.
Giovanni
